From 0ee3545420e0dfa3f75d520b66ea8a31a910fc3d Mon Sep 17 00:00:00 2001
From: Brock Noland <brock@apache.org>
Date: Sun, 1 Mar 2015 19:08:50 +0000
Subject: [PATCH 0060/1164] HIVE-9823 - Load spark-defaults.conf from
 classpath [Spark Branch] (Brock via Xuefu)

git-svn-id: https://svn.apache.org/repos/asf/hive/branches/spark@1663147 13f79535-47bb-0310-9956-ffa450edef68
(cherry picked from commit 160d2dfe0e75cb9db50533f6c3ef80d27cd714c2)
---
 .../apache/hive/spark/client/SparkClientImpl.java  |   15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/spark-client/src/main/java/org/apache/hive/spark/client/SparkClientImpl.java b/spark-client/src/main/java/org/apache/hive/spark/client/SparkClientImpl.java
index ba08106..44549c7 100644
--- a/spark-client/src/main/java/org/apache/hive/spark/client/SparkClientImpl.java
+++ b/spark-client/src/main/java/org/apache/hive/spark/client/SparkClientImpl.java
@@ -24,12 +24,14 @@
 import com.google.common.base.Throwables;
 import com.google.common.collect.Lists;
 import com.google.common.collect.Maps;
+import com.google.common.io.Resources;
 
 import io.netty.channel.ChannelHandlerContext;
 import io.netty.util.concurrent.GenericFutureListener;
 import io.netty.util.concurrent.Promise;
 
 import java.io.BufferedReader;
+import java.io.ByteArrayInputStream;
 import java.io.File;
 import java.io.FileOutputStream;
 import java.io.IOException;
@@ -39,6 +41,7 @@
 import java.io.Serializable;
 import java.io.Writer;
 import java.net.URI;
+import java.net.URL;
 import java.util.List;
 import java.util.Map;
 import java.util.Properties;
@@ -249,6 +252,18 @@ public void run() {
       }
 
       Properties allProps = new Properties();
+      // first load the defaults from spark-defaults.conf if available
+      try {
+        URL sparkDefaultsUrl = Thread.currentThread().getContextClassLoader().getResource("spark-defaults.conf");
+        if (sparkDefaultsUrl != null) {
+          LOG.info("Loading spark defaults: " + sparkDefaultsUrl);
+          allProps.load(new ByteArrayInputStream(Resources.toByteArray(sparkDefaultsUrl)));
+        }
+      } catch (Exception e) {
+        String msg = "Exception trying to load spark-defaults.conf: " + e;
+        throw new IOException(msg, e);
+      }
+      // then load the SparkClientImpl config
       for (Map.Entry<String, String> e : conf.entrySet()) {
         allProps.put(e.getKey(), conf.get(e.getKey()));
       }
-- 
1.7.9.5

